src/basic_formulas.rs

```diff
@@ -26,7 +26,7 @@ impl EmpiricalFormula {
             return Err(());
         }
         while !string.is_empty() {
-            if string[0] < b'A' || string[0] > b'Z' {
+            if !string[0].is_ascii_uppercase() {
```

Comment:
This method does not exist for a u8

Reply:
That's strange, it seem to exist:
 https://doc.rust-lang.org/std/primitive.u8.html#method.is_ascii_uppercase

Reply:
WHAT!?, uh, I remember that when I was starting in rust that function did not exist (my program failed several times for that) and well I never carefully check the basic types in the documentation, it seems that it was included in version 1.24, oh I started as in 1.20.
Forgive me for that